### DISCUSSION TITLE
Need help getting Firemote cards to show on my HA

### DISCUSSION THREAD

**Discussion by AkoreJepade** [PRProd/HA-Firemote#242] [Q&A] [answered]
New to HA but I have managed to get a lot of my gear up and running with HA. I installed the HA-Firemote under HACS but I can never get past the "On any dashboard, click the +ADD CARD button" "Search by cards for "Firemote Card" and click on it" as I never show a "Firemote Card" when I search for it. HA is showing the Shield Android TV as you can see in the attachment:
![AndroidTV](https://github.com/PRProd/HA-Firemote/assets/43866673/4f731212-d83a-4646-bdd9-e6247f360619)

**Comment by PRProd** [comment]
Hello @SirDigitalKnight!  Welcome!  It's good to have you here!

The starting point is very important.  Cards can only be added to dashboards, so you will need to start by creating a dashboard, or opening an existing dashboard.

For example, I have a few custom dashboards that I made inside of my personal Home Assistant setup right now: (blue arrows)

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/29bf3283-038f-45ee-a2b5-4c1d36acd9b3)

<br><br>

## Example: Create A New Dashboard For Firemote
As an example, let's create a brand new dashboard and add a Firemote card.
 1. Click on **Settings**
 2. Click on **Dashboards**

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/9aefae02-3e04-41ed-a7c0-dbe1c278beeb)

 3. Click the **+ ADD DASHBOARD** button
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/5d094169-296b-43c7-9689-c1f1d6f18200)

 4. Enter a name for your new Dashboard.  As you type, the URL field will automatically populate.  Optionally, you can select an icon for your dashboard as well, but this is not required
 5. Click the **CREATE** button
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/403d311c-1f22-47e6-9054-4b350429f23d)

 6. Your new dashboard should now appear on the left panel, and in your dashboard list.  Click on **OPEN** to see your new dashboard
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/75e8b325-f7c8-40e5-8bb6-b8aca8f33d2e)

Your new dashboard will contain controls for all of your Home Assistant integrations.  Let's edit the new dashboard so that it only contains a Firemote card.

 7. Click the three dots in the upper right hand corner, and select **Edit dashboard**
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b16811bd-3d51-4dfa-9f8c-832533932482)

 8. Turn on the option to **Start with an empty dashboard**
 9. Click on **TAKE CONTROL**
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/c1e4e8c5-4826-49fa-bb95-26605902d7ef)

You will now see an empty dashboard, ready for a Firemote Card to be added.

 10. Click the three dots again in the upper right hand corner, and select **Edit dashboard**

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b16811bd-3d51-4dfa-9f8c-832533932482)

 11. Click on the **+ ADD CARD** button
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/e33b2665-7ff7-440c-b5fa-3bcd6a1e7c86)

 12. Scroll down until you find **Custom: Firemote Card** and click on it
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f294e1f2-55bc-41e4-a45e-400b41d2219e)

 13. On the next screen, configure your Firemote card the way you need it, and click the **SAVE** button
 

Hopefully this will be enough information to get you started.  If not, please let me know and I will assist you further.

Good luck!

**Comment by AkoreJepade** [comment]
Thank you for the detailed explanation. Step 12 is where I can't continue as you can see from my screen shot I do not have said card:

Mine:

![MyCards](https://github.com/PRProd/HA-Firemote/assets/43866673/090ec2f3-2d81-4997-8d83-a20fe8bd494c)

Yours:

![CustomFire](https://github.com/PRProd/HA-Firemote/assets/43866673/03df9d01-cac7-4b72-a052-bb2151407633)

**Reply by PRProd** [reply]
Scroll down
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/d64cc920-c8bd-4d99-b88c-45ac087b9b7b)

**Comment by AkoreJepade** [comment]
Thanks for the quick reply! Here are all the cards I have and below them is Manual and that is it. I must be missing something :-(

![cards1](https://github.com/PRProd/HA-Firemote/assets/43866673/1745ea6d-1e18-4482-b102-dda6dadd0ea6)
![cards2](https://github.com/PRProd/HA-Firemote/assets/43866673/916ddf5d-1dc6-4fb3-aabb-11073209db59)
![cards3](https://github.com/PRProd/HA-Firemote/assets/43866673/3135a4d7-a76e-4466-b39e-0d9289a08c23)
![cards4](https://github.com/PRProd/HA-Firemote/assets/43866673/287de70e-b3f9-44c6-89eb-a2a1ce13dec4)

**Reply by PRProd** [reply]
Strange... I'm not sure why your HACS download(s) wouldn't show up here.  They really should.  Have you tried clearing your browser cache?  Home Assistant cache can be VERY sticky!  ([some light reading here for your amusement](https://github.com/PRProd/HA-Firemote/discussions/74#discussioncomment-5618842))

Try this:
 - Click on the Manual choice
 ![image](https://github.com/PRProd/HA-Firemote/assets/25486599/aaf5cd51-d1a9-4ce1-b0ba-979b7f8ec890)

 - This screen will pop up
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/fe333dc1-2eab-4e04-94ef-2fa075414f12)

 - Replace everything on the left side with this:
 ![image](https://github.com/PRProd/HA-Firemote/assets/25486599/86cb11c6-4a30-4882-833b-2484893acf58)
 ```YAML
type: custom:firemote-card
entity: media_player.fire_tv_192_168_1_30
device_family: amazon-fire
device_type: fire_tv_4_series
compatibility_mode: default
scale: 100
 ```
 - Click the **SHOW VISUAL EDITOR** button to guide you from here
 
If this does work, then congratulations... but I don't know why.  :laughing:  If it doesn't work, I would try one or more of these things next:
 1. Force refresh your browser's cache and/or launch a new browser window in incognito/private mode
 2. Restart your Home Assistant
 3. Attempt to re-download Firemote from HACS

**Comment by AkoreJepade** [comment]
![Cardnoexist](https://github.com/PRProd/HA-Firemote/assets/43866673/e18ddb53-1a78-4796-838f-187c0568b83c)

**Reply by PRProd** [reply]
It looks like you'll have to move on to my next three suggestions:
> - Force refresh your browser's cache and/or launch a new browser window in incognito/private mode
> - Restart your Home Assistant
> - Attempt to re-download Firemote from HACS

**Reply by AkoreJepade** [reply]
I followed your 3 steps in order

**Comment by AkoreJepade** [comment]
I now have A card but it does not look like yours?:

![CustomFire](https://github.com/PRProd/HA-Firemote/assets/43866673/104563d7-b06a-4456-ac76-6834846c9cab)

I get this message when I try to set it up:

![needto](https://github.com/PRProd/HA-Firemote/assets/43866673/ed243602-6612-4586-b86f-edad967e99d4)

**Reply by PRProd** [reply]
You haven't defined an Android Debug Bridge Entity in your config.

From the [documentation](https://github.com/PRProd/HA-Firemote#how-to-use):

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/28a19e65-72e6-4bf6-921a-a8ec9d9c4f7e)

<br>

If you don't have any options in that dropdown menu, go back to [these instructions](https://github.com/PRProd/HA-Firemote#download-and-setup-via-home-assistant-ui--hacs) in the documentation:
<br>

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b80b3996-1625-47e1-9c56-bfd602ec4072)

**Reply by AkoreJepade** [reply]
Yeah I think this is the part I cannot get working. I have Developer options enabled on my ShieldTV and I have USB debugging turned on but there is no "ADB Debugging" setting, only and option to disable ADB Auth timeout.? Is there a walk-through YouTube vid or easy to follow steps about this "Android Debug Bridge Intergration"? I think this is the part I cannot seem to get working. :-(

Thank you again for all your help with this!

**Reply by PRProd** [reply]
@SirDigitalKnight - No problem!  I'm on a mission to get this working for you now!

Turn these two options on:
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/a3efce6e-8e4c-4e47-a21d-79be7200d9c5)

Take note of your Shield's IP address listed by the first arrow (yours will be different from mine)  This is the address you will use when you  set up the [Android Debug Bridge Integration](https://www.home-assistant.io/integrations/androidtv/)

**Reply by AkoreJepade** [reply]
Fails to connect

![FailToConnect](https://github.com/PRProd/HA-Firemote/assets/43866673/f5621f6d-41ad-4eec-bc87-9bed0ed34f52)

**Reply by AkoreJepade** [reply]
My shield settings:

![20231108_214401](https://github.com/PRProd/HA-Firemote/assets/43866673/55ac93c4-624f-4b24-92b2-60c6759ba19e)

**Reply by PRProd** [reply, marked as answer]
When you first tried connecting, did you see a message appear on your Shield?  You have to confirm that it's OK for your Home Assistant to connect to your shield when that appears.

**Reply by AkoreJepade** [reply]
Yeah I should have known that from all my Android phone ROMs I've installed over the years :-)
 
![success](https://github.com/PRProd/HA-Firemote/assets/43866673/15cb9668-e676-4044-b6c3-d48ccbeb3e9e)

**Reply by AkoreJepade** [reply]
Is there any way to fix the lag/delay in the remote operation or is that delay caused because it is going through the HA server? The Shield TV app and Google TV app do not have this delay or lag effect.

**Reply by PRProd** [reply]
Based on [your first post](https://github.com/PRProd/HA-Firemote/discussions/242#discussion-5829097), I know that you already have configured an [Android TV Remote](https://www.home-assistant.io/integrations/androidtv_remote/) entity.  Selecting that entity from the **Associated Android TV Remote Entity: (optional)** dropdown menu will give you the speed increase that you are looking for.  

From the [documentation](https://github.com/PRProd/HA-Firemote#how-to-use):
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f98d2141-f5bf-4dd6-8ee4-950bc9122c2e)

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/73a17c2c-801c-4b7b-9ead-0a8bd71e73aa)

**Reply by AkoreJepade** [reply]
Thank you for all the help Doug. I really appreciate it! I will mess around with it this evening (Front Projector Cine so only use at night) and see if I can tweak it to perfection. Can I use one of the other remotes besides the plain Shield TV remote? 

![Screenshot_20231109_130900](https://github.com/PRProd/HA-Firemote/assets/43866673/9e67ce20-8111-4e93-8357-9a17210a8979)

**Reply by PRProd** [reply]
The Firemote card allows you to choose any remote style through this menu:

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f916c838-a57f-423c-a1b8-b3b5e71d9551)

From the [documentation](https://github.com/PRProd/HA-Firemote#how-to-use):
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/7212279d-f58d-4bc5-a6b1-d94c001ef9bc)


I strongly encourage you to read through the [README.md](https://github.com/PRProd/HA-Firemote#readme) document one more time.  There is a lot of information included there that should help answer any additional questions that you might have.  I'm also here to help.

**Reply by AkoreJepade** [reply]
Cool! That is what I was looking for! Thanks again for everything Doug!
